deepspeed/runtime/engine.py

```diff
@@ -2503,14 +2503,20 @@ def load_moe_state_dict(checkpoint_path,
                         state_dict.update(expert_state_dict)
                     moe_layer_id += 1
 
-    def load_module_state_dict(self, checkpoint, strict=True, custom_load_fn=None):
-        module_state_dict = checkpoint['module']
-        if custom_load_fn:
-            custom_load_fn(src=module_state_dict, dst=self.module)
-        else:
-            self.module.load_state_dict(
-                module_state_dict,  # TODO
-                strict=strict)
+    def load_module_state_dict(self, checkpoint, strict=True, custom_load_fn=None, fetch_z3_params=False):
+
+        def z3_params_to_fetch(param_list):
+            return [p for p in param_list if hasattr(p, 'ds_id') and p.ds_status == ZeroParamStatus.NOT_AVAILABLE]
+
+        params_to_fetch = z3_params_to_fetch(self.module.parameters()) if fetch_z3_params else []
```

Comment:
Ternary conditions are often more difficult to read, potentially better alternative:

```suggestion
        def z3_params_to_fetch(param_list):
            if not fetch_z3_params:
                return []
            return [p for p in param_list if hasattr(p, 'ds_id') and p.ds_status == ZeroParamStatus.NOT_AVAILABLE]

        params_to_fetch = z3_params_to_fetch(self.module.parameters())
```

Reply:
I will rewrite differently.